tests/Collection/CrudSpecFunctionalTest.php

```diff
@@ -224,6 +231,38 @@ private function executeAssertResult(array $operation, $expectedResult, $actualR
                 }
                 break;
 
+            case 'bulkWrite':
+                if (isset($expectedResult['deletedCount'])) {
+                    $this->assertSame($expectedResult['deletedCount'], $actualResult->getDeletedCount());
+                }
+
+                if (isset($expectedResult['insertedIds'])) {
+                    $this->assertSameDocument(
+                        ['insertedIds' => $expectedResult['insertedIds']],
+                        ['insertedIds' => $actualResult->getInsertedIds()]
+                    );
+                }
+
+                if (isset($expectedResult['matchedCount'])) {
+                    $this->assertSame($expectedResult['matchedCount'], $actualResult->getMatchedCount());
+                }
+
+                if (isset($expectedResult['modifiedCount'])) {
+                    $this->assertSame($expectedResult['modifiedCount'], $actualResult->getModifiedCount());
+                }
+
+                if (isset($expectedResult['upsertedCount'])) {
+                    $this->assertSame($expectedResult['upsertedCount'], $actualResult->getUpsertedCount());
+                }
+
+                if (array_key_exists('upsertedId', $expectedResult)) {
```

Comment:
No change needed here, but if you're curious as to why we use `array_key_exists()` here instead of `isset()`, it's because `null` is technically a valid `_id` value and could be reported from an upsert (if the user originally supplied the `_id` value themselves, since the server would always use an ObjectId).